gcp/observability/config.go

```diff
@@ -72,21 +74,32 @@ func validateFilters(config *configpb.ObservabilityConfig) error {
 	return nil
 }
 
+// unmarshalConfig unmarshals a json string representing an observability config
+// into it's protobuf format.
+func unmarshalConfig(rawJSON json.RawMessage) (*configpb.ObservabilityConfig, error) {
+	var config configpb.ObservabilityConfig
+	if err := protojson.Unmarshal(rawJSON, &config); err != nil {
+		return nil, fmt.Errorf("error parsing observability config: %v", err)
+	}
+	if err := validateFilters(&config); err != nil {
+		return nil, fmt.Errorf("error parsing observability config: %v", err)
+	}
+	if config.GlobalTraceSamplingRate > 1 || config.GlobalTraceSamplingRate < 0 {
+		return nil, fmt.Errorf("error parsing observability config: invalid global trace sampling rate %v", config.GlobalTraceSamplingRate)
+	}
+	logger.Infof("Parsed ObservabilityConfig: %+v", &config)
+	return &config, nil
+}
+
 func parseObservabilityConfig() (*configpb.ObservabilityConfig, error) {
-	// Parse the config from ENV var
-	if content := os.Getenv(envObservabilityConfig); content != "" {
-		var config configpb.ObservabilityConfig
-		if err := protojson.Unmarshal([]byte(content), &config); err != nil {
-			return nil, fmt.Errorf("error parsing observability config from env %v: %v", envObservabilityConfig, err)
-		}
-		if err := validateFilters(&config); err != nil {
-			return nil, fmt.Errorf("error parsing observability config: %v", err)
-		}
-		if config.GlobalTraceSamplingRate > 1 || config.GlobalTraceSamplingRate < 0 {
-			return nil, fmt.Errorf("error parsing observability config: invalid global trace sampling rate %v", config.GlobalTraceSamplingRate)
+	if fileSystemPath := os.Getenv(envObservabilityConfigJSON); fileSystemPath != "" {
+		content, err := os.ReadFile(fileSystemPath)
+		if err != nil {
+			return nil, fmt.Errorf("error reading file from env var %v: %v", envObservabilityConfigJSON, err)
```

Comment:
Print `fileSystemPath`?  Also avoid shorthand/abbreviations in user-visible error messages ("env var" -> "environment variable").

```go
"error reading observability configuration file %q: %v", fileSystemPath, err"
```